src/main/java/com/google/devtools/build/lib/analysis/BuildView.java

```diff
@@ -545,6 +563,7 @@ public ActionAnalysisMetadata getGeneratingAction(Artifact artifact) {
         aspects,
         allTargetsToTest == null ? null : ImmutableList.copyOf(allTargetsToTest),
         ImmutableSet.copyOf(targetsToSkip),
+        ImmutableSet.copyOf(failedTargets.build()),
```

Comment:
Yes. With my current patch `failedTargets` is a subset of `targetsToSkip`. That was the only way I could get bazel to skip the target and at the same time mark them as failed. Otherwise they were marked with NO STATUS or something like that. I couldn't find a way to communicate both bits of information with what was in AnalysisResult.

The consequence is that the post-build logic has to change. It bucketed everything into passed and skipped. Everything that was not in passed and not skipped was considered failed. But what if skipped targets are also failed? I might be misremembering some of the details. I'll add more comments in places and commit message to help future me and others with this.